exchanges/retry/src/retryExchange.ts

```diff
@@ -97,18 +106,30 @@ export const retryExchange = ({
       forward,
       share,
       filter(res => {
-        const maxNumberAttemptsExceeded =
-          (res.operation.context.retryCount || 0) >= MAX_ATTEMPTS - 1;
         // Only retry if the error passes the conditional retryIf function (if passed)
         // or if the error contains a networkError
-        if (res.error && retryIf(res.error) && !maxNumberAttemptsExceeded) {
+        if (!res.error || !retryIf(res.error)) {
+          return true;
+        }
+
+        const maxNumberAttemptsExceeded =
+          (res.operation.context.retryCount || 0) >= MAX_ATTEMPTS - 1;
+
+        if (!maxNumberAttemptsExceeded) {
           // Send failed responses to be retried by calling next on the retry$ subject
           // Exclude operations that have been retried more than the specified max
           nextRetryOperation(res.operation);
           return false;
-        } else {
-          return true;
         }
+
+        client.debugTarget!.dispatchEvent({
+          type: 'retryMaxAttemptsReached',
```

Comment:
Minor suggestion - how would you feel about the term _exhausted_?

```suggestion
          type: 'retryExhausted',
```